src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.LightweightOverloadResolution.cs

```diff
@@ -5,297 +5,252 @@
+            // find the parameter at the cursor position
+            var argumentIndexToSave = TryGetArgumentIndex();
+            var foundParameterIndex = -1;
+            if (argumentIndexToSave >= 0)
+            {
+                foundParameterIndex = argumentToParameterMap[argumentIndexToSave];
+                if (foundParameterIndex < 0)
+                    foundParameterIndex = FirstUnspecifiedParameter(ref argumentToParameterMap.AsRef());
+            }
 
-                // find the parameter at the cursor position
-                var argumentIndexToSave = TryGetArgumentIndex(arguments, position);
-                if (argumentIndexToSave >= 0)
+            Debug.Assert(foundParameterIndex < parameters.Length);
+
+            return (method, foundParameterIndex);
+        }
+
+        /// <summary>
+        /// Determines if the given argument is compatible with the given parameter
+        /// </summary>
+        private bool IsCompatibleArgument(ArgumentSyntax argument, IParameterSymbol parameter)
+        {
+            var parameterRefKind = parameter.RefKind;
+            if (parameterRefKind == RefKind.None)
+            {
+                if (IsEmptyArgument(argument.Expression))
                 {
-                    var foundParam = argToParamMap[argumentIndexToSave];
-                    if (foundParam >= 0)
-                    {
-                        foundParameterIndex = foundParam;
-                    }
-                    else
-                    {
-                        foundParameterIndex = FirstUnspecifiedParameter(argToParamMap, argumentCount);
-                    }
+                    // An argument left empty is considered to match any parameter
+                    // M(1, $$)
+                    // M(1, , 2$$)
+                    return true;
                 }
-                else
+
+                var type = parameter.Type;
+                if (parameter.IsParams
+                    && type is IArrayTypeSymbol arrayType
+                    && semanticModel.ClassifyConversion(argument.Expression, arrayType.ElementType).IsImplicit)
                 {
-                    foundParameterIndex = -1;
+                    return true;
                 }
 
-                Debug.Assert(foundParameterIndex < parameters.Length);
+                return semanticModel.ClassifyConversion(argument.Expression, type).IsImplicit;
+            }
 
+            var argumentRefKind = argument.GetRefKind();
+            if (parameterRefKind == argumentRefKind)
                 return true;
 
-                // If the cursor is pointing at an argument for which we did not find the corresponding
-                // parameter, we will highlight the first unspecified parameter.
-                static int FirstUnspecifiedParameter(ArrayBuilder<int> argToParamMap, int argumentCount)
-                {
-                    using var _ = ArrayBuilder<bool>.GetInstance(argumentCount, false, out var specified);
```

Comment:
this logic was very suspect.  it made an array filled with 'false', but which had a count of 0.  so this would only work properly if the argToParameterIndex happened to be in order.  if, say, there was an out of order arg/param, this would index improperly.